docs/ref-manual/approx/array-hashes.md

```diff
@@ -0,0 +1,44 @@
+Array hashes
+============
+
+### Background: Non-Collision of Hashes is Essential for Storage Integrity
```

Comment:
Is the storage integrity issue related to the approximation?  It's hard to tell without the example.  Does the hashing approximation mean that if I write to `m["some really long string"]` I just havoc everything because memory integrity is violated?